### PR TITLE
[BISERVER-12695] - Repository REST service to rename file returns HTT…

### PR DESCRIPTION
**Pull request by yshakhau**
…P 500 even though the action appears to succeed anyway

### CODE REVIEW

File: extensions/src/org/pentaho/platform/web/http/api/resources/FileResource.java [hunk 1]
```diff
@@ -2235,4 +2241,13 @@ protected Document parseText( String text ) throws DocumentException {
   protected Messages getMessagesInstance() {
     return Messages.getInstance();
   }
+  
+  private void checkCorrectExtension(String fileName) {
+    IRepositoryContentConverterHandler handler = PentahoSystem.get( IRepositoryContentConverterHandler.class );
+    String ext = RepositoryFilenameUtils.getExtension( fileName );
+
+    if ( handler != null && handler.getConverter( ext ) == null ) {
+      throw new IllegalArgumentException( Messages.getInstance().getString( "FileResource.INCORRECT_EXTENTION", fileName ) );
```
Comment (rpbouman) [inline extensions/src/org/pentaho/platform/web/http/api/resources/FileResource.java]: EXTENSION is the correct spelling.

File: extensions/src/org/pentaho/platform/web/http/messages/messages.properties [hunk 2]
```diff
@@ -142,4 +142,6 @@ XactionUtil.XML_OUTPUT_NOT_SUPPORTED=Unable to execute xaction as an xml output
 SystemResource.GENERAL_ERROR=Error getting system resource
 
 HsqlDatabaseStartupListener.ERROR_0001_HSQLDB_ENTRY_MALFORMED=SQL to get databases is malformed
-HsqldbStartupListener.ERROR_0004_INVALID_PORT=Invalid port specified. Defaulting to {0}
\ No newline at end of file
+HsqldbStartupListener.ERROR_0004_INVALID_PORT=Invalid port specified. Defaulting to {0}
+
+FileResource.INCORRECT_EXTENTION={0} has incorrect extension.
```
Comment (rpbouman) [inline extensions/src/org/pentaho/platform/web/http/messages/messages.properties]: -> EXTENSION

File: extensions/test-src/org/pentaho/platform/web/http/api/resources/FileResourceTest.java [hunk 3]
```diff
@@ -58,8 +64,15 @@
 import static org.mockito.Mockito.*;
 
 public class FileResourceTest {
-
-  FileResource fileResource;
+  private static final String XML_EXTENTION = "xml";
```
Comment (rpbouman) [inline extensions/test-src/org/pentaho/platform/web/http/api/resources/FileResourceTest.java]: EXTENSION

**Comment by yshakhau** [comment]
Spelling fixed.

**Pull request merged**